usr/share/rear/finalize/Fedora/i386/550_rebuild_initramfs.sh

```diff
@@ -47,31 +47,26 @@ if [ -f $VAR_DIR/recovery/initrd_modules ]; then
     OLD_INITRD_MODULES=()
 fi
 
-Log "Original OLD_INITRD_MODULES='${OLD_INITRD_MODULES[@]}'"
-# To see what has been added by the migration process, the new modules are added to the
-# end of the list. To achieve this, we list the old modules twice in the variable
-# NEW_INITRD_MODULES and then add the new modules. Then we use "uniq -u" to filter out
-# the modules which only appear once in the list. The resulting array
-# contains the new modules also.
+Log "Original OLD_INITRD_MODULES=( ${OLD_INITRD_MODULES[*]} )"
+# To see what has been added by the migration process, the new modules are added to the end of the list.
+# To achieve this, we list the old modules twice in the variable NEW_INITRD_MODULES and then add the new modules.
+# Then we use "uniq -u" to filter out the modules which only appear once in the list.
+# The resulting array contains only the new modules:
 NEW_INITRD_MODULES=( ${OLD_INITRD_MODULES[@]} ${OLD_INITRD_MODULES[@]} $( cat $TMP_DIR/storage_drivers ) )
```

Comment:
I believe that it would be better to quote `${OLD_INITRD_MODULES[@]}`. Not that it matters here (module names can't contain whitespace anyway, otherwise the code would break later), but I find it preferable for consistency with places where it actually matters. ShellCheck does not complain about the unquoted array expansion, btw?

Reply:
Done via
https://github.com/rear/rear/commit/90930fa7168be63fe6d1b668f870b03451c6a141

ShellCheck shows the unquoted array expansion only as a warining
```
# shellcheck -s bash -S warning usr/share/rear/finalize/Fedora/i386/550_rebuild_initramfs.sh
...
NEW_INITRD_MODULES=( ${OLD_INITRD_MODULES[@]} ${OLD_INITRD_MODULES[@]} $( cat $TMP_DIR/storage_drivers ) )
                     ^----------------------^ SC2206: Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
                                              ^----------------------^ SC2206: Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.
                                                                       ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
```
but not as an error with
```
shellcheck -s bash -S error usr/share/rear/finalize/Fedora/i386/550_rebuild_initramfs.sh
```
and currently I am only interested in what is considered to be an error by ShellCheck
cf. https://github.com/rear/rear/pull/2771#issuecomment-1067997675
because we have 685 ShellCheck reported "errors" where 571 SC2168
are likely all false positives so we have about 100 real ShellCheck "errors"
but we have 2192 ShellCheck reported "warnings" that include 685 "errors"
so about 1500 ShellCheck "only warnings" that is far too much for me,
cf. https://github.com/rear/rear/issues/1040#issuecomment-1062945160